netsim/modules/evpn.py

```diff
@@ -114,6 +114,55 @@ def vlan_aware_bundle_service(vlan: Box, vname: str, topology: Box) -> None:
   evpn.vlans.append(vname)                                          # Finally, add VLAN name to the list of MAC VRF VLANs
   evpn.vlan_ids.append(topology.vlans[vname].id)                    # ... and a VLAN ID to list of EVPN-enabled VLAN tags
 
+
+"""
+Figure out whether we have to treat a VLAN as part of a VLAN bundle or an independent VLAN service
+"""
+
+def get_vlan_bundle_flag(vlan: Box, vname: str, topology: Box) -> bool:
+  if not 'vrf' in vlan:                                             # VLAN not part of a VRF ==> VLAN service
+    return False
+
+  vrf = topology.vrfs[vlan.vrf]
+  if not data.get_from_box(vrf,'evpn.bundle'):                      # VRF does not have evpn.bundle attribute ==> VLAN service
+    return False
+
+  must_be_string(
+    parent=vrf,
+    key='evpn.bundle',
+    path=f'vrfs.{vlan.vrf}',
+    module='evpn',
+    valid_values=topology.defaults.evpn.attributes.bundle)          # Validate evpn.bundle value
+
+  return True
+
+"""
+Create VLAN services -- either as independent VLANs or as a VRF-based bundle
+"""
+
+def create_vlan_service(vname: str, topology: Box) -> None:
+  vlan = topology.vlans[vname]
+  if not get_vlan_bundle_flag(vlan,vname,topology):
+    vlan_based_service(vlan,vname,topology)
+  else:
+    vlan_aware_bundle_service(vlan,vname,topology)
```

Comment:
This method is now a bit of a misnomer - ```vlan_bundle_service``` would be more accurate

Reply:
In theory, you're correct. In practice, let's wait and see whether we need anything more than what this one contains (when you implement other bundle services).